python/cudf/cudf/core/scalar.py

```diff
@@ -125,15 +310,15 @@ def __init__(self, value, dtype=None):
             else:
                 self._device_value = value._device_value
         else:
-            self._host_value, self._host_dtype = self._preprocess_host_value(
+            self._host_value, self._host_dtype = _preprocess_host_value(
                 value, dtype
             )
 
     @classmethod
-    def from_device_scalar(cls, device_scalar):
-        if not isinstance(device_scalar, cudf._lib.scalar.DeviceScalar):
+    def from_device_scalar(cls, device_scalar: plc.Scalar) -> Self:
```

Comment:
I can't remember if the plan is to eventually remove `cudf.Scalar`? But this could be
```suggestion
    def from_pylibcudf(cls, scalar: plc.Scalar) -> Self:
```

Reply:
Agreed this is better, thanks!